homeassistant/components/amcrest/__init__.py

```diff
@@ -143,7 +153,19 @@ def setup(hass, config):
                 CONF_NAME: name,
             }, config)
 
+        if binary_sensors:
+            discovery.load_platform(
+                hass, 'binary_sensor', DOMAIN, {
+                    CONF_NAME: name,
+                    CONF_BINARY_SENSORS: binary_sensors
+                }, config)
+
         if sensors:
+            if 'motion_detector' in sensors:
```

Comment:
Add a custom validator that checks for deprecation. See mysensors component for example.

Reply:
Out of curiosity, why does mysensors have its own implementation of a deprecation validator when one already exists in config_validation.py? Also, the one in mysensors removes the deprecated key. That doesn't seem right. In my understanding, when something is deprecated it means it is still supported but will eventually not be supported. So removing the key would be wrong. It should just be issuing a warning.

I can add a custom deprecation validator, but it will only issue a warning.

Reply:
Yes, implement it to fit the need here. I mentioned mysensors since it's using similar custom validators.

Reply:
It the mysensors case, we still "support" the key by removing it, instead of erroring.

The mysensors validator predates the common deprecation validator.

Reply:
I did implement it. Did you see the changes?